devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -202,7 +204,10 @@ public void execute() throws MojoExecutionException {
             // resolve the outcome we need here
             RunnerJarOutcome result = appCreator.resolveOutcome(RunnerJarOutcome.class);
             Artifact original = project.getArtifact();
-            original.setFile(result.getOriginalJar().toFile());
+            Path originalJar = result.getOriginalJar();
```

Comment:
I'm a bit confused here.

If finalName is used why `getOriginalJar` is `null`, it should be the `finalName`. If not created, what t the project main artifact?

Reply:
> If finalName is used why `getOriginalJar` is `null`, it should be the `finalName`. 

I expected the same behaviour which is the case when set using Maven `finalName` build property. However this is not the case when `finalName` is a configuration property in `quarkus-maven-plugin`. 

> If not created, what t the project main artifact?

I am sorry, I did not get you here.

Reply:
Could we remove our specific configuration? @aloubyansky 
That's in the context of Gradle and native CLI support too I suppose.

Reply:
I actually was not aware we did it this way, I don't like this ".original" trick at all. The original JAR should not be replaced or renamed just because we decided to generate an extra JAR that is NOT conflicting with the original JAR.

Reply:
Ah, I see they are replacing it if the name is the same, sorry. I guess I would just move `originalJar = outputDir.resolve(finalName + ".jar");` before the `if(uberJar)`. That way originalJar would never be null.

Reply:
@emmanuelbernard I guess we could. There is a difference though between the Maven `finalName` and the one configured for `quarkus-maven-plugin`. If, e.g. the Maven `finalName` is initialized to `original` and the `quarkus-maven-plugin` to `quarkus` then `mvn package` will produce two JARs: `original.jar` and `other-runner.jar`.
So our `finalName` is affecting only the runner JAR filename, while the Maven one - the original app JAR.
TBH, I don't very much like what our `finalName` does. I'd prefer `runnerName` instead or something.

Reply:
...the value of which should be the complete runner JAR file name, w/o expecting the -runner suffix be appended by the mojo.

Reply:
@aloubyansky thanks for the comments.

> Ah, I see they are replacing it if the name is the same, sorry. I guess I would just move `originalJar = outputDir.resolve(finalName + ".jar");` before the `if(uberJar)`. That way originalJar would never be null.

This should work. I'll make an update.

> ...the value of which should be the complete runner JAR file name, w/o expecting the -runner suffix be appended by the mojo.

+1.

Reply:
> ...the value of which should be the complete runner JAR file name, w/o expecting the -runner suffix be appended by the mojo.

@aloubyansky would we want to include this a part of the PR?

Reply:
No. But I think you can create a separate issue for it.

Reply:
I think it is related to this issue https://github.com/quarkusio/quarkus/issues/2931.

Reply:
Yes, pretty confusing. IMO, we should get rid of the ".original" one.
I don't think it makes sense attaching a non-uberjar to the project neither. It's useless w/o the lib.

Reply:
> Ah, I see they are replacing it if the name is the same, sorry. I guess I would just move `originalJar = outputDir.resolve(finalName + ".jar");` before the `if(uberJar)`. That way originalJar would never be null.

@aloubyansky, comment above has been addressed.